tag_release.py

```diff
@@ -0,0 +1,554 @@
+            "sha": commit_info['sha'],
+        }
+        resp = session.post(ref_url, json=payload)
+        if resp.ok:
+            succeeded.append(repo_name)
+        else:
+            failed_resp = resp
+            failed_repo = repo_name
+            # don't try to tag any others, just stop
+            break
+
+    if failed_resp is None:
+        return True
+
+    # if we got to this point, then there was a failure.
+    try:
+        original_err_msg = failed_resp.json()["message"]
+    except Exception:
+        original_err_msg = failed_resp.text
+
+    if not succeeded:
+        msg = (
+            "Failed to create {ref_name} on {failed_repo}. "
+            "Error was {orig_err}. No tags have been created on any repos."
+        ).format(
+            ref_name=ref_name,
+            failed_repo=failed_repo,
+            orig_err=original_err_msg,
+        )
+        log.error(msg)
+        return False
+
+    if rollback_on_fail:
+        rollback_failures = []
+        for repo_name in succeeded:
+            ref_url = "https://api.github.com/repos/{repo}/git/{ref_name}".format(
+                repo=repo_name,
+                ref_name=ref_name,
+            )
+            resp = session.delete(ref_url)
+            if not resp.ok:
+                rollback_failures.append(repo_name)
+
+        if rollback_failures:
+            msg = (
+                "Failed to create {ref_name} on {failed_repo}. "
+                "Error was {orig_err}. "
+                "Attempted to roll back, but failed to delete tag on "
+                "the following repos: {rollback_failures}"
+            ).format(
+                ref_name=ref_name,
+                failed_repo=failed_repo,
+                orig_err=original_err_msg,
+                rollback_failures=", ".join(rollback_failures)
+            )
+            err = RuntimeError(msg)
+            err.response = failed_resp
+            err.repos = rollback_failures
+            raise err
+        else:
+            msg = (
+                "Failed to create {ref_name} on {failed_repo}. "
+                "Error was {orig_err}. However, all refs were successfully "
+                "rolled back."
+            ).format(
+                ref_name=ref_name,
+                failed_repo=failed_repo,
+                orig_err=original_err_msg,
+            )
+            log.error(msg)
+            return False
+    else:
+        # don't try to rollback, just raise an error
+        msg = (
+            "Failed to create {ref_name} on {failed_repo}. "
+            "Error was {orig_err}. No rollback attempted. Tags exist on "
+            "the following repos: {tagged_repos}"
+        ).format(
+            ref_name=ref_name,
+            failed_repo=failed_repo,
+            orig_err=original_err_msg,
+            tagged_repos=", ".join(succeeded)
+        )
+        err = RuntimeError(msg)
+        err.response = failed_resp
+        err.repos = succeeded
```

Comment:
No. It's just so that if another Python script calls these functions in the future, the data is more easily accessed than trying to parse the message.